### ISSUE TITLE
Mill Library log duplication.

### ISSUE THREAD

**Issue by ShawnieBoy** [magarena/magarena#264]
Triggered abilities with text describing a mill action (e.g. Belltower Sphinx) Cause a duplication of the ability on the log as the MillLibraryAction also feeds to the log when it is performed.

**Comment by melvinzhang** [comment]
Most actions don't add a log messages on what they do, unless it is to clarify the steps. Perhaps we should remove the log message in MillLibraryAction.

**Comment by melvinzhang** [comment]
For example, DestroyAction is one where is makes sense to log a message in the case something else occurred, such as the permanent is regenerated, has a totem shield etc.

**Comment by ShawnieBoy** [comment]
I agree, was surprised to see it as I would expect it from an event.

**Comment by melvinzhang** [comment]
Just did some digging the log has been there all the while. My guess is that it is added to show the actual number of cards moved compared to what was intended due to the library having in sufficient cards. So perhaps a more succinct message just indicating the amount of cards will do.

**Comment by ShawnieBoy** [comment]
Maybe as a logAppend to prevent the creation of what appears to be another event?

**Comment by melvinzhang** [comment]
Definitely.

**Comment by ShawnieBoy** [comment]
While on the subject, would it make sense for DamageAction to players to have some kind of log feedback (Barring combat damage to players, which is already put onto the log)

**Comment by melvinzhang** [comment]
What's an example of the log feedback for damage to player?

**Comment by ShawnieBoy** [comment]
PN is dealt <amount> damage.

Mainly for abilities and effects that deal damage to all players.

**Comment by ShawnieBoy** [comment]
Looking at it, it is quite niche, I ended up including log info on Wing Storm as there was no log info for the damage dealt

**Comment by melvinzhang** [comment]
Wouldn't it replicate the event? Eg

Lightning bolt deals 3 damage to target player or creature. (PlayerName). PlayerName is dealt 3 damage.

**Comment by ShawnieBoy** [comment]
It's mainly for things where the damage dealt isn't fixed, or is determined elsewhere. More likely something that needs to be done in groovy than in the engine

**Comment by ShawnieBoy** [comment]
Like [Acidic Soil](http://magiccards.info/us/en/172.html)

**Comment by melvinzhang** [comment]
Will be easy to miss if done in groovy. Now that we have a concept of getting a number from the game state via MagicAmount. How about a constructor of DealDamage that takes in a MagicAmount instead of an int. In that case, it will output a log message on the amount. 

Something like:
```
new DealDamageAction(event.getSource(), it, {
  source -> CREATURE_WITH_FLYING_YOU_CONTROL.filter(it).size() * 2
})
```

**Comment by melvinzhang** [comment]
Or even simpler is a factory method that always logs,
```
DealDamageAction.Variable(
  event.getSource(), 
  it, 
  CREATURE_WITH_FLYING_YOU_CONTROL.filter(it).size() * 2
)
```

**Comment by ShawnieBoy** [comment]
That sounds a good idea!
